src/api/app/models/staging_workflow.rb

```diff
@@ -20,10 +20,12 @@ def ignored_requests
 
   private
 
-  def init_staging_projects
-    return unless new_record?
+  def create_staging_projects
     ['A', 'B'].each do |letter|
-      staging_projects << Project.find_or_initialize_by(name: "#{project.name}:Staging:#{letter}")
+      staging_project = Project.find_or_initialize_by(name: "#{project.name}:Staging:#{letter}")
+      next if staging_project.staging_workflow # if it belongs to another staging workflow skip it
```

Comment:
you want multiple staging workflows per project? Do you have other hurting hobbies as well?

Reply:
Nope! But is would be possible (but unlikely) to manually create ``openSUSE:Factory:Staging:A`` and assign it to ``openSUSE:Leap``s staging workflow. In this case we should skip assigning it to the new staging workflow and just leave it as it is.